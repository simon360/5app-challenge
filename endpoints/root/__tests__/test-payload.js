const parsePayload = require("../payload");
const parseItem = require("../item");

// For this test, swap out parseItem for the identity function.
// We don't want to test parseItem's functionality by mistake.
jest.mock("../item", () => jest.fn(item => item));

describe("payload", () => {
  it("should map each filtered item through parseItem", () => {
    parsePayload([
      {
        count: 3
      },
      {
        count: 14
      },
      {
        count: 1
      }
    ]);

    // Two calls. Should not be called when count > 1
    expect(parseItem).toHaveBeenCalledTimes(2);
  });

  it("should remove items with counts <= 1", () => {
    expect(
      parsePayload([
        {
          count: 0
        },
        {
          count: -1
        },
        {
          count: 1
        },
        {
          count: 2
        }
      ])
    ).toMatchSnapshot();
  });
});
