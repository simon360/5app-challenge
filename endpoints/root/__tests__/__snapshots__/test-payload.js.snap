// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`payload should remove items with counts <= 1 1`] = `
Array [
  Object {
    "count": 2,
  },
]
`;
